### PR TITLE
netbsd adding _lwp_park api.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 1]
```diff
@@ -2862,6 +2862,22 @@ extern "C" {
...
+    #[link_name = "___lwp_park60"]
+    pub fn _lwp_park(
+        clock: ::clockid_t,
+        flags: ::c_int,
+        ts: *const ::timespec,
+        unpark: ::lwpid_t,
+        hint: *const ::c_void,
+        unparkhint: *mut ::c_void,
+    ) -> ::c_int;
+    pub fn _lwp_unpark(id: ::lwpid_t, hint: *const ::c_void) -> ::c_int;
```
Comment (tgross35) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:2875]: Optional nit: `id` is called `lwp` per https://man.netbsd.org/_lwp_unpark.2

File: src/unix/bsd/netbsdlike/netbsd/mod.rs [hunk 2]
```diff
@@ -2862,6 +2862,22 @@ extern "C" {
     pub fn getrandom(buf: *mut ::c_void, buflen: ::size_t, flags: ::c_uint) -> ::ssize_t;
 
     pub fn reboot(mode: ::c_int, bootstr: *mut ::c_char) -> ::c_int;
+
+    #[link_name = "___lwp_park60"]
```
Comment (tgross35) [inline src/unix/bsd/netbsdlike/netbsd/mod.rs:2866]: Do you have a link/source referencing the link name? I can't seem to find it mentioned anywhere

**Comment by tgross35** [comment]
One question and one optional nit, otherwise lgtm. Relevant sources:

- https://man.netbsd.org/_lwp_park.2
- https://man.netbsd.org/_lwp_unpark.2
- https://man.netbsd.org/_lwp_unpark_all.2

@rustbot author

File: src/unix/bsd/netbsdlike/netbsd/mod.rs (hunk 2 above)
Comment (devnexen) [reply src/unix/bsd/netbsdlike/netbsd/mod.rs:2866]: [here for example](https://github.com/NetBSD/src/blob/889f3bb010ad20d396fb291b89f202288dac2c87/include/lwp.h#L57)

Comment (devnexen) [reply src/unix/bsd/netbsdlike/netbsd/mod.rs:2866]: note that rust uses directly [the "real" symbol](https://github.com/rust-lang/rust/blob/e9e27ab0cf3759e420eeaa6bab7c7d454d9c10cd/library/std/src/sys/pal/unix/thread_parking.rs#L33).

Comment (tgross35) [reply src/unix/bsd/netbsdlike/netbsd/mod.rs:2866]: Perfect, thank you.

**Pull request merged**